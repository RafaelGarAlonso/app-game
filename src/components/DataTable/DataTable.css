thead, tr, th {
    width: 100%;
    display: flex;
    flex-direction: row;
}

th, td {
    display: flex;
    flex: 1;
    justify-content: center;
    cursor: pointer;
}