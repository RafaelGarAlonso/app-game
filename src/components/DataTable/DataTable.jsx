import { Row, Col, Table } from 'react-bootstrap';
import './DataTable.css';

export const DataTable = (props) => {
    return(
        <Row>
            <Col>
                <Table className="table" size="sm" striped bordered>     
                    <thead>
                        <tr>
                        {
                            props.headers.map( (head, index) => {
                                return <th key = { index } >{head}</th>
                            })
                        }
                        </tr>
                    </thead>
                    <tbody>
                        {
                            props.rows.length ? props.rows.map( (row, index) => {
                                return <tr id={row.uid} key = { index }>
                                    <td>{ index + 1 }</td>
                                    <td>{ row.userName }</td>
                                    <td>{ row.points }</td>
                                </tr>
                            })
                            : 
                            <tr>
                                <td>EMPTY LIST</td>
                            </tr>
                        }
                    </tbody>
                </Table>
            </Col>
        </Row>
    )
}