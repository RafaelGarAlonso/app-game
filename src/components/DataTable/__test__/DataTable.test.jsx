import { render } from '@testing-library/react';
import { DataTable } from '../DataTable';

const DATA_TABLE_HEADERS = ['#', 'Username', 'Points'];

const INIT_USERS_TABLE = [
  {
    userName: "Manolo",
    points: 100
  },
  {
    userName: "Pepe",
    points: 88
  },
  {
    userName: "Benito",
    points: 50
  }
];

describe('Test <DataTable />', () => {

  test('Check snapshot component', () => {
    const { container } = render(<DataTable headers={DATA_TABLE_HEADERS} rows={INIT_USERS_TABLE} />);
    expect(container).toMatchSnapshot();
    ;
  });

  test('Check that data is rendered', () => {
    const { container } = render(<DataTable headers={DATA_TABLE_HEADERS} rows={INIT_USERS_TABLE} />);
    const head01 = container.querySelectorAll("th")[1].textContent;
    const head02 = container.querySelectorAll("th")[2].textContent;
    const row01 = container.querySelectorAll("td")[1].textContent;
    const row02 = container.querySelectorAll("td")[2].textContent;

    expect(head01).toBe(DATA_TABLE_HEADERS[1]);
    expect(head02).toBe(DATA_TABLE_HEADERS[2]);
    expect(row01).toBe(INIT_USERS_TABLE[0].userName);
    expect(row02).toBe(INIT_USERS_TABLE[0].points.toString());
    ;
  });

  test('Check that data is rendered', () => {
    const { container } = render(<DataTable headers={DATA_TABLE_HEADERS} rows={[]} />);
    const emptyList = container.querySelector("tbody").textContent;
    expect(emptyList).toBe("EMPTY LIST");
  });

});