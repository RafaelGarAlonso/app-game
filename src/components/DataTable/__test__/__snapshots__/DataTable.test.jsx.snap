// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <DataTable /> Check snapshot component 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <table
        class="table table table-sm table-striped table-bordered"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Username
            </th>
            <th>
              Points
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              Manolo
            </td>
            <td>
              100
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              Pepe
            </td>
            <td>
              88
            </td>
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              Benito
            </td>
            <td>
              50
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
